extractBuildAndPush.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Contract:
+# $context = the extraction root
+# $gitsource = the target application, plus reference to docker image
+# $dockerfile = the target image name
+# Steps:
+# 1. Go to $gitsource. Parse the config yaml to get the source image
+# 2. Pull the image
+# 3. Spin up a container out of it
+# 4. Copy the extracted content from the container mount to $context
+# 5. Copy the git source to $context/user-app
+# 5. Re-build the image, using the Docker file rom the source image
+# 6. Push the image
+# 7. Delete the container, and the image.
+
+mkdir -p $context
+stack=`cat $context/../git-source/.appsody-config.yaml | awk -F' ' '{print $2}'`
+buildah pull $stack
+buildah from --name base-extract $stack
+cid=`buildah ps -aq`
+iid=`buildah images -aq`
+mount=`buildah mount base-extract`
+cp -rf $mount/project/* $context
+cp -rf $context/../$gitsource/* $context/user-app
```

Comment:
IMO, one thing that needs deeper inspection / deliberation is the above 2 lines - static assumptions about:
 -  the appsody project vortex being always at `/project` 
 - the github project vortex being always place-able under `user-app` under the context

This may break some stacks; the question is:  does it make sense to define such a contract and adapt all projects to it.

Reply:
Re: /project does https://github.com/appsody/appsody/pull/190 help with your concern?
@neeraj-laad do you have any thoughts re: user-app?

Reply:
Couple of things: 
1) There's now support for customizing the location of the appsody project through the `APPSODY_PROJECT_DIR` variable.
2) The location of the user's code within that context is dictated by whatever is specified by the creator of the stack in the `APPSODY_MOUNTS` env var. Which - BTW - now needs to "jive" with the `APPSODY_PROJECT_DIR`.

Reply:
Yes, APPSODY_PROJECT_DIR should be used to determine the project root folder, and it does not seem right to enforce `user-app` as the location for user's app. We have a few stacks that do not follow this structure already.

Reply:
@neeraj-laad - on this topic, since `APPSODY_MOUNTS` references the root project folder, should we require (or "strongly suggest") that all references to the project root folder are made through `APPSODY_PROJECT_DIR` if it's defined? I can see situations where someone sets `APPSODY_PROJECT_DIR` to `/xyz`, but then the mounts reference `/abc/src`. That would cause `appsody extract` to fail now.